PR addressing Issue #43

Max-Bladen opened the pull request:
After extensive exploration, it was determined that the `overall` error rates were consistent across the two functions (`perf.mint.splsda()` and `tune.mint.splsda()`). Hence, only the `BER` calculation needed to be rectified. 

The inconsistency was caused by the way the global BER was calculated in each function:

- `tune.mint.splsda()`: Calculated the confusion matrix of a model *for each study* and then found the mean of these study-specific BER values
- `perf.mint.splsda()`: Calculated the confusion matrix of a model on all samples, irrespective of study. A single BER was yielded from this.

The `perf.mint.splsda()` had its functionality changed such that it was brought in line with the way in which `tune.mint.splsda()` calculated global BER. 

The way in which `global$overall` is calculated in `perf.mint.splsda()` was only adjusted for consistency with `global$BER` - the outputted values in the previous `overall` method and new `overall` method are the same.

A minor addition was made to BOTH functions (in the case of `tune.mint.splsda()`, this occured in `LOGOCV()`). Rather than the global BER being just a basic average of each study-specific BER, both now find the average BER across studies, weighted by the study sample sizes

Max-Bladen committed:
Fix for Issue #43

aljabadi commented:
Thanks @Max-Bladen.

Can you please add a unit test that tests for this issue, where the test fails before the fix and passes after it? If this behaviour might be affecting other functions, it would be great to add tests for those as well.

Max-Bladen committed:
Added tests to `test-perf.mint.splsda.R`

Included a test to check that the error rate values yielded by `perf.mint.splsda()` and `tune.mint.splsda()` are the same (to 4 decimal places). This is checked for all three distance metrics, using both standard and balanced error rates.

Max-Bladen commented:
@aljabadi latest commit adds a test which checks for homogenity between the two methods for all distance (`max.dist`, `centroid.dist` and `mahalanobis.dist`) and error rate (`overall` and `BER`) metrics.
